### ISSUE TITLE
Refining creative-mode templates for Sept 2023 release

### ISSUE THREAD

**Issue by colleenXu** [biothings/biothings_explorer#461]
Making a new issue to track the work related to writing creative-model query templates (involves testing them against diseases in individual queries, and deciding the order to run them in....). 

Related to previous comments in #449 and #453.

**Comment by colleenXu** [comment]
TDLR: There's still work:
* Does the corrected notebook need to run on the latest DrugMechDB data?
* Review short metapaths from that notebook for use as templates (cells 48, 51?) Max of 3 [qEdges](https://github.com/biothings/BioThings_Explorer_TRAPI/issues/449#issue-1247208784)
* Write / refine templates. Remember design so that individual results within them == unique drugs (using is_set parameter to do this).

From Slack discussions:
* Andrew wanted me to start with Carolina's [templates](https://github.com/Carolina1396/drug_mechanisms_rare_diseases_BioThingsExplorer/tree/main/src/query_templates), assuming that these were related to an older DrugMechDB analysis (output of cells 20-21 [here](https://github.com/SuLab/DMDB_Analysis/commit/4a6c48e95df4ff452c3906557366cdc21ff0918e)). However, these two didn't actually match each other
* Turns out Carolina's templates were based on Kevin's bte-regression-testing [here](https://github.com/biothings/bte_regression_test/tree/main/src/query_templates), which doesn't seem to be based on DrugMechDB analysis
* Carolina ran a corrected DrugMechDB analysis on a "DrugMechDB version downloaded in April". I stored it [here](https://github.com/colleenXu/RegistryMetadataDev/commit/6c1fab54e67e66c4a4fdd1ce2b2141a2a6f9142f)
    * What Carolina said about the notebook: The results from this notebook are wrong because the sequence of meta-nodes are extracted incorrectly from the  DMDB path file. For example, in a path that only has one biological process identifier,  the extracted meta-path shows two biological process nodes.

**Comment by andrewsu** [comment]
I'm going to close https://github.com/biothings/BioThings_Explorer_TRAPI/issues/237, but just noting here that orthology-based templates could also be good for drug-disease templates for creative mode...

**Comment by tokebe** [comment]
@colleenXu @andrewsu Just taking a brief browse through the TAQA triage board, looks like [this issue](https://github.com/NCATSTranslator/Ordering-Organizing/issues/21) may come from us. I think we could avoid this with some changes to make the directionality of `m4-Chem-Gene-DoP.json` more specific (possibly through qualifiers/etc). Though, based on @andrewsu's [comment in another issue](https://github.com/NCATSTranslator/Feedback/issues/139#issuecomment-1442459575), it appears we may not want to, if I understand?

**Comment by colleenXu** [comment]
Seems like @andrewsu would be the one to decide...I agree that adding a qualifier constraint to the [template](https://github.com/biothings/bte_trapi_query_graph_handler/blob/41d12d4db6aaab057f6fcd29114d74f8f73f95fa/data/templates/Drug-treats-Disease/m4-Chem-Gene-DoP.json#L20) could work to make this more specific (object direction is decreased)

**Comment by tokebe** [comment]
Based on today's meeting it seems we want to make the template more specific, but duplicate the less-specific version as a template with lower execution priority.

**Comment by colleenXu** [comment]
Keep in mind: UI has a 3-hop limit and subclass edges from node-expansion will count as 1. So templates should be only 1 / 2 hops right now...

**Comment by colleenXu** [comment]
Also, can modify templates now that "qualifier hierarchy expansion" is now supported. See https://github.com/biothings/biothings_explorer/issues/598

**Comment by colleenXu** [comment]
As discussed in today's meeting, the scope of this issue (defined for Sept release) is:
* [make the directionality of m4-Chem-Gene-DoP.json more specific](https://github.com/biothings/biothings_explorer/issues/461#issuecomment-1568934102) 
* [simplify templates now that qualifier expansion is supported](https://github.com/biothings/biothings_explorer/issues/461#issuecomment-1642889993)

**Comment by colleenXu** [comment]
Originally, this issue was addressed by https://github.com/biothings/bte_trapi_query_graph_handler/pull/167. 

But after #699 and #701 were identified, I adjusted the templates. [The two main asks of this issue](https://github.com/biothings/biothings_explorer/issues/461#issuecomment-1681031438) are still addressed by the adjusted templates + templateGroups. 

Details of the adjustments made can be seen in #699 and #701 and in the PR that addresses them https://github.com/biothings/bte_trapi_query_graph_handler/pull/169#issuecomment-1683004440. I include basic info on the latest testing in this PR.

**Comment by tokebe** [comment]
Tracking will continue in linked and future issues, closing this one as completed
